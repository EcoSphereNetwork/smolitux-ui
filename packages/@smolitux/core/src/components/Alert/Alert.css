.smolitux-alert {
  position: relative;
  padding: 1rem;
  border-radius: 0.375rem;
  display: flex;
  align-items: flex-start;
  width: 100%;
  overflow: hidden;
}

.smolitux-alert--success {
  background-color: var(--alert-success-bg, #ecfdf5);
  border-left: 4px solid var(--alert-success-border, #10b981);
  color: var(--alert-success-text, #065f46);
}

.smolitux-alert--error {
  background-color: var(--alert-error-bg, #fef2f2);
  border-left: 4px solid var(--alert-error-border, #ef4444);
  color: var(--alert-error-text, #991b1b);
}

.smolitux-alert--warning {
  background-color: var(--alert-warning-bg, #fffbeb);
  border-left: 4px solid var(--alert-warning-border, #f59e0b);
  color: var(--alert-warning-text, #92400e);
}

.smolitux-alert--info {
  background-color: var(--alert-info-bg, #eff6ff);
  border-left: 4px solid var(--alert-info-border, #3b82f6);
  color: var(--alert-info-text, #1e40af);
}

.smolitux-alert__icon {
  flex-shrink: 0;
  margin-right: 0.75rem;
  width: 1.25rem;
  height: 1.25rem;
}

.smolitux-alert__content {
  flex: 1;
}

.smolitux-alert__title {
  font-weight: 600;
  font-size: 1rem;
  margin-bottom: 0.25rem;
}

.smolitux-alert__message {
  font-size: 0.875rem;
}

.smolitux-alert__close {
  position: absolute;
  top: 0.5rem;
  right: 0.5rem;
  background: transparent;
  border: none;
  cursor: pointer;
  padding: 0.25rem;
  display: flex;
  align-items: center;
  justify-content: center;
  border-radius: 0.25rem;
  color: currentColor;
  opacity: 0.6;
  transition: opacity 0.2s ease-in-out;
}

.smolitux-alert__close:hover {
  opacity: 1;
}

.smolitux-alert__close:focus {
  outline: 2px solid currentColor;
  outline-offset: 1px;
}

.smolitux-alert__actions {
  margin-top: 0.75rem;
}

.smolitux-alert__progress {
  position: absolute;
  bottom: 0;
  left: 0;
  height: 3px;
  background-color: currentColor;
  opacity: 0.3;
  transition: width 0.1s linear;
}

/* Animationen */
@keyframes smolitux-alert-fade-in {
  from {
    opacity: 0;
    transform: translateY(-10px);
  }
  to {
    opacity: 1;
    transform: translateY(0);
  }
}

@keyframes smolitux-alert-fade-out {
  from {
    opacity: 1;
    transform: translateY(0);
  }
  to {
    opacity: 0;
    transform: translateY(-10px);
  }
}

.smolitux-alert--animate-in {
  animation: smolitux-alert-fade-in 0.3s ease-out forwards;
}

.smolitux-alert--animate-out {
  animation: smolitux-alert-fade-out 0.3s ease-in forwards;
}

/* Dark Mode */
.dark .smolitux-alert--success {
  background-color: var(--alert-success-bg-dark, rgba(16, 185, 129, 0.2));
  color: var(--alert-success-text-dark, #34d399);
}

.dark .smolitux-alert--error {
  background-color: var(--alert-error-bg-dark, rgba(239, 68, 68, 0.2));
  color: var(--alert-error-text-dark, #f87171);
}

.dark .smolitux-alert--warning {
  background-color: var(--alert-warning-bg-dark, rgba(245, 158, 11, 0.2));
  color: var(--alert-warning-text-dark, #fbbf24);
}

.dark .smolitux-alert--info {
  background-color: var(--alert-info-bg-dark, rgba(59, 130, 246, 0.2));
  color: var(--alert-info-text-dark, #60a5fa);
}