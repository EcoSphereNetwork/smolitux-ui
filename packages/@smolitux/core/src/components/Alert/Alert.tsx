import React, { useRef, useEffect } from 'react';

export type AlertType = 'success' | 'error' | 'warning' | 'info';

export interface AlertProps {
  /** Typ der Benachrichtigung */
  type: AlertType;
  /** Titel der Benachrichtigung */
  title?: string;
  /** Nachrichtentext */
  message: string | React.ReactNode;
  /** Callback zum Schließen der Benachrichtigung */
  onClose?: () => void;
  /** Zusätzliche CSS-Klassen */
  className?: string;
  /** Icon anzeigen */
  showIcon?: boolean;
  /** Schließbar machen */
  closable?: boolean;
  /** Automatisch schließen nach X Millisekunden (0 = nicht automatisch schließen) */
  autoClose?: number;
  /** Zusätzlicher Inhalt (z.B. Aktions-Buttons) */
  children?: React.ReactNode;
  /** ID für Barrierefreiheit */
  id?: string;
  /** Ob der Alert beim Laden fokussiert werden soll (für Screenreader) */
  autoFocus?: boolean;
}

/**
 * Alert-Komponente für Feedback und Benachrichtigungen
 * 
 * @example
 * ```tsx
 * <Alert type="success" title="Erfolg" message="Die Aktion wurde erfolgreich ausgeführt." />
 * <Alert type="error" message="Es ist ein Fehler aufgetreten." closable onClose={handleClose} />
 * <Alert type="warning" title="Achtung" message="Diese Aktion kann nicht rückgängig gemacht werden." />
 * <Alert type="info" message="Die Wartungsarbeiten beginnen in 5 Minuten." autoClose={5000} />
 * ```
 */
export const Alert: React.FC<AlertProps> = ({
  type,
  title,
  message,
  onClose,
  className = '',
  showIcon = true,
  closable = false,
  autoClose = 0,
  children,
  id,
  autoFocus = false
}) => {
  const alertRef = useRef<HTMLDivElement>(null);
  const alertId = id || `alert-${Math.random().toString(36).substr(2, 9)}`;
  const titleId = `${alertId}-title`;
  const messageId = `${alertId}-message`;
  
  // Auto-Close Funktionalität
  useEffect(() => {
    let timeoutId: NodeJS.Timeout;
    
    if (autoClose > 0 && onClose) {
      timeoutId = setTimeout(() => {
        onClose();
      }, autoClose);
    }
    
    return () => {
      if (timeoutId) {
        clearTimeout(timeoutId);
      }
    };
  }, [autoClose, onClose]);
  
  // Auto-Focus für Screenreader
  useEffect(() => {
    if (autoFocus && alertRef.current) {
      alertRef.current.focus();
    }
  }, [autoFocus]);

  // Stil-Varianten basierend auf dem Typ
  const typeStyles = {
    success: {
      background: 'bg-green-50 dark:bg-green-900/20',
      border: 'border-green-300 dark:border-green-800',
      titleColor: 'text-green-800 dark:text-green-300',
      textColor: 'text-green-700 dark:text-green-400',
      icon: (
        <svg className="w-5 h-5 text-green-500" fill="none" stroke="currentColor" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg" aria-hidden="true">
          <path strokeLinecap="round" strokeLinejoin="round" strokeWidth="2" d="M9 12l2 2 4-4m6 2a9 9 0 11-18 0 9 9 0 0118 0z"></path>
        </svg>
      ),
      ariaLive: 'polite'
    },
    error: {
      background: 'bg-red-50 dark:bg-red-900/20',
      border: 'border-red-300 dark:border-red-800',
      titleColor: 'text-red-800 dark:text-red-300',
      textColor: 'text-red-700 dark:text-red-400',
      icon: (
        <svg className="w-5 h-5 text-red-500" fill="none" stroke="currentColor" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg" aria-hidden="true">
          <path strokeLinecap="round" strokeLinejoin="round" strokeWidth="2" d="M10 14l2-2m0 0l2-2m-2 2l-2-2m2 2l2 2m7-2a9 9 0 11-18 0 9 9 0 0118 0z"></path>
        </svg>
      ),
      ariaLive: 'assertive'
    },
    warning: {
      background: 'bg-yellow-50 dark:bg-yellow-900/20',
      border: 'border-yellow-300 dark:border-yellow-800',
      titleColor: 'text-yellow-800 dark:text-yellow-300',
      textColor: 'text-yellow-700 dark:text-yellow-400',
      icon: (
        <svg className="w-5 h-5 text-yellow-500" fill="none" stroke="currentColor" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg" aria-hidden="true">
          <path strokeLinecap="round" strokeLinejoin="round" strokeWidth="2" d="M12 9v2m0 4h.01m-6.938 4h13.856c1.54 0 2.502-1.667 1.732-3L13.732 4c-.77-1.333-2.694-1.333-3.464 0L3.34 16c-.77 1.333.192 3 1.732 3z"></path>
        </svg>
      ),
      ariaLive: 'polite'
    },
    info: {
      background: 'bg-blue-50 dark:bg-blue-900/20',
      border: 'border-blue-300 dark:border-blue-800',
      titleColor: 'text-blue-800 dark:text-blue-300',
      textColor: 'text-blue-700 dark:text-blue-400',
      icon: (
        <svg className="w-5 h-5 text-blue-500" fill="none" stroke="currentColor" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg" aria-hidden="true">
          <path strokeLinecap="round" strokeLinejoin="round" strokeWidth="2" d="M13 16h-1v-4h-1m1-4h.01M21 12a9 9 0 11-18 0 9 9 0 0118 0z"></path>
        </svg>
      ),
      ariaLive: 'polite'
    }
  };

  const styles = typeStyles[type];
  
  // Keyboard-Handler für Schließen-Button
  const handleKeyDown = (e: React.KeyboardEvent) => {
    if (e.key === 'Escape' && onClose) {
      onClose();
    }
  };

  return (
    <div 
      ref={alertRef}
      id={alertId}
      className={`p-4 border rounded-lg ${styles.background} ${styles.border} ${className}`} 
      role="alert"
      aria-live="polite"
      aria-labelledby={title ? titleId : undefined}
      aria-describedby={messageId}
      tabIndex={autoFocus ? 0 : undefined}
      onKeyDown={handleKeyDown}
    >
      <div className="flex">
        {showIcon && (
          <div className="flex-shrink-0">
            {styles.icon}
          </div>
        )}
        <div className={`${showIcon ? 'ml-3' : ''} flex-1`}>
          {title && (
            <h3 id={titleId} className={`text-sm font-medium ${styles.titleColor}`}>{title}</h3>
          )}
          <div id={messageId} className={`text-sm ${styles.textColor} ${title ? 'mt-2' : ''}`}>
            {message}
          </div>
          {children && <div className="mt-3">{children}</div>}
        </div>
        {(closable || onClose) && (
          <button
            type="button"
            className={`ml-auto -mx-1.5 -my-1.5 ${styles.textColor} hover:bg-gray-100 dark:hover:bg-gray-800 rounded-lg p-1.5 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-primary-500`}
            onClick={onClose}
            aria-label="Close"
          >
            <span className="sr-only">Close</span>
            <svg className="w-5 h-5" fill="none" stroke="currentColor" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg" aria-hidden="true">
              <path strokeLinecap="round" strokeLinejoin="round" strokeWidth="2" d="M6 18L18 6M6 6l12 12"></path>
            </svg>
          </button>
        )}
      </div>
    </div>
  );
};

export default Alert;