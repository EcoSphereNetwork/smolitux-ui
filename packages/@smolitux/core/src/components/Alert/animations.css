/* Fade-In Animation */
@keyframes alertFadeIn {
  from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
}

.alert-animate-fadeIn {
  animation: alertFadeIn 0.3s ease-in-out forwards;
}

/* Fade-Out Animation */
@keyframes alertFadeOut {
  from {
    opacity: 1;
  }
  to {
    opacity: 0;
  }
}

.alert-animate-fadeOut {
  animation: alertFadeOut 0.3s ease-in-out forwards;
}

/* Slide-In-Right Animation */
@keyframes alertSlideInRight {
  from {
    opacity: 0;
    transform: translateX(20px);
  }
  to {
    opacity: 1;
    transform: translateX(0);
  }
}

.alert-animate-slideInRight {
  animation: alertSlideInRight 0.3s ease-in-out forwards;
}

/* Slide-Out-Right Animation */
@keyframes alertSlideOutRight {
  from {
    opacity: 1;
    transform: translateX(0);
  }
  to {
    opacity: 0;
    transform: translateX(20px);
  }
}

.alert-animate-slideOutRight {
  animation: alertSlideOutRight 0.3s ease-in-out forwards;
}

/* Slide-In-Down Animation */
@keyframes alertSlideInDown {
  from {
    opacity: 0;
    transform: translateY(-20px);
  }
  to {
    opacity: 1;
    transform: translateY(0);
  }
}

.alert-animate-slideInDown {
  animation: alertSlideInDown 0.3s ease-in-out forwards;
}

/* Slide-Out-Down Animation */
@keyframes alertSlideOutDown {
  from {
    opacity: 1;
    transform: translateY(0);
  }
  to {
    opacity: 0;
    transform: translateY(20px);
  }
}

.alert-animate-slideOutDown {
  animation: alertSlideOutDown 0.3s ease-in-out forwards;
}