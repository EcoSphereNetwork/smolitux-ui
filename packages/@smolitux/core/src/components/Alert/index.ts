export { Alert } from './Alert';
export type { AlertProps, AlertType } from './Alert';

// Für Abwärtskompatibilität mit dem bestehenden Export
export { Alert as default } from './Alert';