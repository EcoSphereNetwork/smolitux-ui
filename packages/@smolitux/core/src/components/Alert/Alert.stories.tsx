import React from 'react';
import { Meta, StoryObj } from '@storybook/react';
import Alert from './Alert';

const meta: Meta<typeof Alert> = {
  title: 'Core/Alert',
  component: Alert,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
  argTypes: {
    type: {
      control: { type: 'select' },
      options: ['success', 'error', 'warning', 'info'],
    },
    variant: {
      control: { type: 'select' },
      options: ['default', 'outline', 'filled', 'subtle'],
    },
    animation: {
      control: { type: 'select' },
      options: ['fade', 'slide-right', 'slide-down', 'none'],
    },
    title: { control: 'text' },
    message: { control: 'text' },
    showIcon: { control: 'boolean' },
    closable: { control: 'boolean' },
    autoClose: { control: 'number' },
    autoFocus: { control: 'boolean' },
    animated: { control: 'boolean' },
    compact: { control: 'boolean' },
    bordered: { control: 'boolean' },
    rounded: { control: 'boolean' },
    shadow: { control: 'boolean' },
  },
};

export default meta;
type Story = StoryObj<typeof Alert>;

export const Success: Story = {
  args: {
    type: 'success',
    title: 'Erfolg',
    message: 'Die Aktion wurde erfolgreich ausgeführt.',
    showIcon: true,
  },
};

export const Error: Story = {
  args: {
    type: 'error',
    title: 'Fehler',
    message: 'Es ist ein Fehler aufgetreten.',
    showIcon: true,
  },
};

export const Warning: Story = {
  args: {
    type: 'warning',
    title: 'Achtung',
    message: 'Diese Aktion kann nicht rückgängig gemacht werden.',
    showIcon: true,
  },
};

export const Info: Story = {
  args: {
    type: 'info',
    title: 'Information',
    message: 'Die Wartungsarbeiten beginnen in 5 Minuten.',
    showIcon: true,
  },
};

export const WithoutTitle: Story = {
  args: {
    type: 'info',
    message: 'Dies ist eine Benachrichtigung ohne Titel.',
    showIcon: true,
  },
};

export const Closable: Story = {
  args: {
    type: 'info',
    title: 'Schließbar',
    message: 'Klicken Sie auf das X, um diese Benachrichtigung zu schließen.',
    showIcon: true,
    closable: true,
    onClose: () => console.log('Alert geschlossen'),
  },
};

export const OutlineVariant: Story = {
  args: {
    type: 'success',
    title: 'Outline-Variante',
    message: 'Dies ist eine Benachrichtigung mit Outline-Variante.',
    showIcon: true,
    variant: 'outline',
  },
};

export const FilledVariant: Story = {
  args: {
    type: 'error',
    title: 'Filled-Variante',
    message: 'Dies ist eine Benachrichtigung mit Filled-Variante.',
    showIcon: true,
    variant: 'filled',
  },
};

export const SubtleVariant: Story = {
  args: {
    type: 'warning',
    title: 'Subtle-Variante',
    message: 'Dies ist eine Benachrichtigung mit Subtle-Variante.',
    showIcon: true,
    variant: 'subtle',
  },
};

export const WithActions: Story = {
  args: {
    type: 'info',
    title: 'Mit Aktionen',
    message: 'Möchten Sie fortfahren?',
    showIcon: true,
    actions: [
      { label: 'Abbrechen', onClick: () => console.log('Abbrechen geklickt') },
      { label: 'Fortfahren', onClick: () => console.log('Fortfahren geklickt'), variant: 'primary' },
    ],
  },
};

export const Compact: Story = {
  args: {
    type: 'info',
    message: 'Dies ist eine kompakte Benachrichtigung.',
    showIcon: true,
    compact: true,
  },
};

export const WithShadow: Story = {
  args: {
    type: 'success',
    title: 'Mit Schatten',
    message: 'Diese Benachrichtigung hat einen Schatten.',
    showIcon: true,
    shadow: true,
  },
};

export const WithoutIcon: Story = {
  args: {
    type: 'warning',
    title: 'Ohne Icon',
    message: 'Diese Benachrichtigung hat kein Icon.',
    showIcon: false,
  },
};

export const WithCustomContent: Story = {
  args: {
    type: 'info',
    title: 'Mit benutzerdefiniertem Inhalt',
    message: 'Diese Benachrichtigung enthält benutzerdefinierten Inhalt.',
    showIcon: true,
    children: (
      <div className="mt-2 p-2 bg-blue-100 dark:bg-blue-900/30 rounded">
        <p className="text-sm text-blue-700 dark:text-blue-300">
          Hier können Sie beliebigen Inhalt hinzufügen.
        </p>
      </div>
    ),
  },
};