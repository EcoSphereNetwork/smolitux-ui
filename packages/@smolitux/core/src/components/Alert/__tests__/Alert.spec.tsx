import React from 'react';
import renderer from 'react-test-renderer';
import { Alert } from '../Alert';

describe('Alert Snapshots', () => {
  test('renders correctly with default props', () => {
    const tree = renderer.create(<Alert type="info" message="Test message" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with title', () => {
    const tree = renderer.create(<Alert type="info" title="Test Title" message="Test message" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with success type', () => {
    const tree = renderer.create(<Alert type="success" message="Success message" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with error type', () => {
    const tree = renderer.create(<Alert type="error" message="Error message" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with warning type', () => {
    const tree = renderer.create(<Alert type="warning" message="Warning message" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly without icon', () => {
    const tree = renderer.create(<Alert type="info" message="Test message" showIcon={false} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with close button', () => {
    const tree = renderer.create(<Alert type="info" message="Test message" closable />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with children', () => {
    const tree = renderer.create(
      <Alert type="info" message="Test message">
        <button>Action Button</button>
      </Alert>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with custom className', () => {
    const tree = renderer.create(<Alert type="info" message="Test message" className="custom-class" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with ReactNode as message', () => {
    const tree = renderer.create(
      <Alert 
        type="info" 
        message={<span data-testid="custom-message">Custom Message</span>} 
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});