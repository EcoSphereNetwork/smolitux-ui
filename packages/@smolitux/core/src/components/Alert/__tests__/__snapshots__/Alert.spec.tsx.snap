// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert Snapshots renders correctly with ReactNode as message 1`] = `
<div
  aria-describedby="alert-6wuxbfjtb-message"
  aria-live="polite"
  className="p-4 border rounded-lg bg-blue-50 dark:bg-blue-900/20 border-blue-300 dark:border-blue-800 "
  id="alert-6wuxbfjtb"
  onKeyDown={[Function]}
  role="alert"
>
  <div
    className="flex"
  >
    <div
      className="flex-shrink-0"
    >
      <svg
        aria-hidden="true"
        className="w-5 h-5 text-blue-500"
        fill="none"
        stroke="currentColor"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M13 16h-1v-4h-1m1-4h.01M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
        />
      </svg>
    </div>
    <div
      className="ml-3 flex-1"
    >
      <div
        className="text-sm text-blue-700 dark:text-blue-400 "
        id="alert-6wuxbfjtb-message"
      >
        <span
          data-testid="custom-message"
        >
          Custom Message
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Alert Snapshots renders correctly with children 1`] = `
<div
  aria-describedby="alert-06lkyup8a-message"
  aria-live="polite"
  className="p-4 border rounded-lg bg-blue-50 dark:bg-blue-900/20 border-blue-300 dark:border-blue-800 "
  id="alert-06lkyup8a"
  onKeyDown={[Function]}
  role="alert"
>
  <div
    className="flex"
  >
    <div
      className="flex-shrink-0"
    >
      <svg
        aria-hidden="true"
        className="w-5 h-5 text-blue-500"
        fill="none"
        stroke="currentColor"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M13 16h-1v-4h-1m1-4h.01M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
        />
      </svg>
    </div>
    <div
      className="ml-3 flex-1"
    >
      <div
        className="text-sm text-blue-700 dark:text-blue-400 "
        id="alert-06lkyup8a-message"
      >
        Test message
      </div>
      <div
        className="mt-3"
      >
        <button>
          Action Button
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Alert Snapshots renders correctly with close button 1`] = `
<div
  aria-describedby="alert-gmjlveept-message"
  aria-live="polite"
  className="p-4 border rounded-lg bg-blue-50 dark:bg-blue-900/20 border-blue-300 dark:border-blue-800 "
  id="alert-gmjlveept"
  onKeyDown={[Function]}
  role="alert"
>
  <div
    className="flex"
  >
    <div
      className="flex-shrink-0"
    >
      <svg
        aria-hidden="true"
        className="w-5 h-5 text-blue-500"
        fill="none"
        stroke="currentColor"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M13 16h-1v-4h-1m1-4h.01M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
        />
      </svg>
    </div>
    <div
      className="ml-3 flex-1"
    >
      <div
        className="text-sm text-blue-700 dark:text-blue-400 "
        id="alert-gmjlveept-message"
      >
        Test message
      </div>
    </div>
    <button
      aria-label="Close"
      className="ml-auto -mx-1.5 -my-1.5 text-blue-700 dark:text-blue-400 hover:bg-gray-100 dark:hover:bg-gray-800 rounded-lg p-1.5 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-primary-500"
      type="button"
    >
      <span
        className="sr-only"
      >
        Close
      </span>
      <svg
        aria-hidden="true"
        className="w-5 h-5"
        fill="none"
        stroke="currentColor"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6 18L18 6M6 6l12 12"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`Alert Snapshots renders correctly with custom className 1`] = `
<div
  aria-describedby="alert-yclf9o815-message"
  aria-live="polite"
  className="p-4 border rounded-lg bg-blue-50 dark:bg-blue-900/20 border-blue-300 dark:border-blue-800 custom-class"
  id="alert-yclf9o815"
  onKeyDown={[Function]}
  role="alert"
>
  <div
    className="flex"
  >
    <div
      className="flex-shrink-0"
    >
      <svg
        aria-hidden="true"
        className="w-5 h-5 text-blue-500"
        fill="none"
        stroke="currentColor"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M13 16h-1v-4h-1m1-4h.01M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
        />
      </svg>
    </div>
    <div
      className="ml-3 flex-1"
    >
      <div
        className="text-sm text-blue-700 dark:text-blue-400 "
        id="alert-yclf9o815-message"
      >
        Test message
      </div>
    </div>
  </div>
</div>
`;

exports[`Alert Snapshots renders correctly with default props 1`] = `
<div
  aria-describedby="alert-jmy6m4qaf-message"
  aria-live="polite"
  className="p-4 border rounded-lg bg-blue-50 dark:bg-blue-900/20 border-blue-300 dark:border-blue-800 "
  id="alert-jmy6m4qaf"
  onKeyDown={[Function]}
  role="alert"
>
  <div
    className="flex"
  >
    <div
      className="flex-shrink-0"
    >
      <svg
        aria-hidden="true"
        className="w-5 h-5 text-blue-500"
        fill="none"
        stroke="currentColor"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M13 16h-1v-4h-1m1-4h.01M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
        />
      </svg>
    </div>
    <div
      className="ml-3 flex-1"
    >
      <div
        className="text-sm text-blue-700 dark:text-blue-400 "
        id="alert-jmy6m4qaf-message"
      >
        Test message
      </div>
    </div>
  </div>
</div>
`;

exports[`Alert Snapshots renders correctly with error type 1`] = `
<div
  aria-describedby="alert-j0o3p8jr7-message"
  aria-live="polite"
  className="p-4 border rounded-lg bg-red-50 dark:bg-red-900/20 border-red-300 dark:border-red-800 "
  id="alert-j0o3p8jr7"
  onKeyDown={[Function]}
  role="alert"
>
  <div
    className="flex"
  >
    <div
      className="flex-shrink-0"
    >
      <svg
        aria-hidden="true"
        className="w-5 h-5 text-red-500"
        fill="none"
        stroke="currentColor"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10 14l2-2m0 0l2-2m-2 2l-2-2m2 2l2 2m7-2a9 9 0 11-18 0 9 9 0 0118 0z"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
        />
      </svg>
    </div>
    <div
      className="ml-3 flex-1"
    >
      <div
        className="text-sm text-red-700 dark:text-red-400 "
        id="alert-j0o3p8jr7-message"
      >
        Error message
      </div>
    </div>
  </div>
</div>
`;

exports[`Alert Snapshots renders correctly with success type 1`] = `
<div
  aria-describedby="alert-hqjnov9r5-message"
  aria-live="polite"
  className="p-4 border rounded-lg bg-green-50 dark:bg-green-900/20 border-green-300 dark:border-green-800 "
  id="alert-hqjnov9r5"
  onKeyDown={[Function]}
  role="alert"
>
  <div
    className="flex"
  >
    <div
      className="flex-shrink-0"
    >
      <svg
        aria-hidden="true"
        className="w-5 h-5 text-green-500"
        fill="none"
        stroke="currentColor"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9 12l2 2 4-4m6 2a9 9 0 11-18 0 9 9 0 0118 0z"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
        />
      </svg>
    </div>
    <div
      className="ml-3 flex-1"
    >
      <div
        className="text-sm text-green-700 dark:text-green-400 "
        id="alert-hqjnov9r5-message"
      >
        Success message
      </div>
    </div>
  </div>
</div>
`;

exports[`Alert Snapshots renders correctly with title 1`] = `
<div
  aria-describedby="alert-h1ftd0aoh-message"
  aria-labelledby="alert-h1ftd0aoh-title"
  aria-live="polite"
  className="p-4 border rounded-lg bg-blue-50 dark:bg-blue-900/20 border-blue-300 dark:border-blue-800 "
  id="alert-h1ftd0aoh"
  onKeyDown={[Function]}
  role="alert"
>
  <div
    className="flex"
  >
    <div
      className="flex-shrink-0"
    >
      <svg
        aria-hidden="true"
        className="w-5 h-5 text-blue-500"
        fill="none"
        stroke="currentColor"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M13 16h-1v-4h-1m1-4h.01M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
        />
      </svg>
    </div>
    <div
      className="ml-3 flex-1"
    >
      <h3
        className="text-sm font-medium text-blue-800 dark:text-blue-300"
        id="alert-h1ftd0aoh-title"
      >
        Test Title
      </h3>
      <div
        className="text-sm text-blue-700 dark:text-blue-400 mt-2"
        id="alert-h1ftd0aoh-message"
      >
        Test message
      </div>
    </div>
  </div>
</div>
`;

exports[`Alert Snapshots renders correctly with warning type 1`] = `
<div
  aria-describedby="alert-afvorw2i2-message"
  aria-live="polite"
  className="p-4 border rounded-lg bg-yellow-50 dark:bg-yellow-900/20 border-yellow-300 dark:border-yellow-800 "
  id="alert-afvorw2i2"
  onKeyDown={[Function]}
  role="alert"
>
  <div
    className="flex"
  >
    <div
      className="flex-shrink-0"
    >
      <svg
        aria-hidden="true"
        className="w-5 h-5 text-yellow-500"
        fill="none"
        stroke="currentColor"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 9v2m0 4h.01m-6.938 4h13.856c1.54 0 2.502-1.667 1.732-3L13.732 4c-.77-1.333-2.694-1.333-3.464 0L3.34 16c-.77 1.333.192 3 1.732 3z"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
        />
      </svg>
    </div>
    <div
      className="ml-3 flex-1"
    >
      <div
        className="text-sm text-yellow-700 dark:text-yellow-400 "
        id="alert-afvorw2i2-message"
      >
        Warning message
      </div>
    </div>
  </div>
</div>
`;

exports[`Alert Snapshots renders correctly without icon 1`] = `
<div
  aria-describedby="alert-x6wk925gp-message"
  aria-live="polite"
  className="p-4 border rounded-lg bg-blue-50 dark:bg-blue-900/20 border-blue-300 dark:border-blue-800 "
  id="alert-x6wk925gp"
  onKeyDown={[Function]}
  role="alert"
>
  <div
    className="flex"
  >
    <div
      className=" flex-1"
    >
      <div
        className="text-sm text-blue-700 dark:text-blue-400 "
        id="alert-x6wk925gp-message"
      >
        Test message
      </div>
    </div>
  </div>
</div>
`;
